<h2 style = 'color:blue' align ='center'> K_Fold Cross Validation <h2>

In [6]:
import numpy as np
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score

In [2]:
irys = load_iris()

Logistic Regression

In [8]:
lr_score = cross_val_score(LogisticRegression(),irys.data,irys.target)
lr_score

c:\Users\rmaxr\AppData\Local\Programs\Python\Python313\Lib\site-packages\sklearn\linear_model\_logistic.py:465: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [9]:
np.mean(lr_score)

np.float64(0.9733333333333334)

Support Vector Machine (SVM)

In [10]:
svm_score = cross_val_score(SVC(),irys.data,irys.target)
svm_score

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [11]:
np.mean(svm_score)

np.float64(0.9666666666666666)

Random Forest

In [12]:
rf_score = cross_val_score(RandomForestClassifier(n_estimators=40),irys.data,irys.target)
rf_score

array([0.96666667, 0.96666667, 0.93333333, 0.93333333, 1.        ])

In [13]:
np.mean(rf_score)

np.float64(0.96)

Decision Tree

In [14]:
dt_score = cross_val_score(DecisionTreeClassifier(),irys.data,irys.target)
dt_score

array([0.96666667, 0.96666667, 0.9       , 1.        , 1.        ])

In [15]:
np.mean(dt_score)

np.float64(0.9666666666666668)

The best score so far is Logistic Regrssion (0.97)